subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r82
+
+import org.gradle.integtests.tooling.TestLauncherSpec
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.TestExecutionException
+import org.gradle.tooling.TestLauncher
+import org.gradle.tooling.TestSpecs
+import spock.lang.Issue
+
+import static org.gradle.integtests.tooling.fixture.TextUtil.normaliseLineSeparators
+
+@ToolingApiVersion('>=7.6')
+@TargetGradleVersion(">=8.2")
+class TestLauncherTestSpecCrossVersionSpec extends TestLauncherSpec {
+
+    def setup() {
+        withFailingTest() // ensures that withTestsFor statements are not ignored
+    }
+
+    @TargetGradleVersion("=8.1")
```

Comment:
❓ Isn't this method annotation in conflict with the class level annotation?

Reply:
Ok, conceptually though, isn't it odd to have a test targeted at `=8.1` in a class targeted at `>=8.2`?

Reply:
Correct, although I only wanted to prove that we had a broken behavior which we fixed in 8.2. I understand the confusion though.

Reply:
Fixed, ptal.

Reply:
I still see a test seemingly in conflict with the class level annotation but I don't have a feeling strong enough to object.

Reply:
This is a common pattern in cross-version tests. On the class level we define a range of target gradle versions (typically from current to all future versions) and override it on the method level for tests that describe old behavior.